osc-server/src/main/java/org/osc/core/broker/service/persistence/PolicyEntityMgr.java

```diff
@@ -43,11 +47,17 @@ public static Policy findById(EntityManager em, Long id) {
 	 * Verifies if the request contains valid policies supported by security manager available on the OSC.
 	 * If the request contains one or more invalid policies, throw an exception.
 	 */
-	public static Set<Policy> findPoliciesById(EntityManager em, Set<Long> ids) throws VmidcBrokerValidationException, Exception {
+	public static Set<Policy> findPoliciesById(EntityManager em, Set<Long> ids, ApplianceManagerConnector mc)
+			throws VmidcBrokerValidationException, Exception {
 		Set<Policy> policies = new HashSet<>();
 		Set<String> invalidPolicies = new HashSet<>();
 		for (Long id : ids) {
-			Policy policy = em.find(Policy.class, id);
+			CriteriaBuilder cb = em.getCriteriaBuilder();
+			CriteriaQuery<Policy> query = cb.createQuery(Policy.class);
+			Root<Policy> root = query.from(Policy.class);
+			query = query.select(root).where(cb.equal(root.get("id"), id),
```

Comment:
since you are getting the ApplianceManagerConnector  you dont need a .get("id")

root.join("applianceManagerConnector"), mc) should be good.